src/blog/ViewsContainer.js

```diff
@@ -1,69 +1,69 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
 import Slider from 'react-slick'
 import { Container, Row, Column } from '../components/grid'
 import ViewsCard from './ViewsCard'
 import { CardContainer, ViewsTitle } from './ViewsContainer.styles'
 import { settings } from './slickSettings'
 
-class ViewsContainer extends React.Component {
+const ViewsContainer = ({ views }) => {
   // Methods here to prevent horizontal scrolling
   // while swiping with react-slick, see
   // https://github.com/akiran/react-slick/issues/1240
-  componentDidMount() {
-    window.addEventListener('touchstart', this.touchStart)
-    window.addEventListener('touchmove', this.preventTouch, { passive: false })
-  }
 
-  componentWillUnmount() {
-    window.removeEventListener('touchstart', this.touchStart)
-    window.removeEventListener('touchmove', this.preventTouch, {
-      passive: false,
-    })
-  }
+  const [firstClientX, setFirstClientX] = useState()
```

Comment:
Take a look at using `ref` instead of `state`. Every time the state is changed, the component will re-render. This could be very slow on performance as the event may be called multiple times per view.